packages/Php70/src/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector.php

```diff
@@ -0,0 +1,259 @@
+        }
+
+        $node->setAttribute(AttributeKey::SCOPE, $scope);
+
+        return $node;
+    }
+
+    /**
+     * @param FuncCall|MethodCall|StaticCall $node
+     *
+     * @return array<int, Expr>
+     */
+    private function getNonVariableArguments(Node $node, Scope $scope): array
+    {
+        $arguments = [];
+
+        /** @var ParameterReflection $parameter */
+        foreach ($this->getParameters($node, $scope) as $key => $parameter) {
+            // omitted optional parameter
+            if (! isset($node->args[$key])) {
+                continue;
+            }
+
+            if ($parameter->passedByReference()->no()) {
+                continue;
+            }
+
+            $argument = $node->args[$key]->value;
+
+            if ($this->isVariableLikeNode($argument)) {
+                continue;
+            }
+
+            $arguments[$key] = $argument;
+        }
+
+        return $arguments;
+    }
+
+    /**
+     * @param FuncCall|MethodCall|StaticCall $node
+     *
+     * @return array<int, ParameterReflection>
+     */
+    private function getParameters(Node $node, Scope $scope): array
+    {
+        if ($node instanceof MethodCall || $node instanceof StaticCall) {
+            $classType = $this->getObjectType($node instanceof MethodCall ? $node->var : $node->class);
+            $methodName = $this->getName($node->name);
+
+            if ($methodName === null || ! $classType->hasMethod($methodName)->yes()) {
+                return [];
+            }
+
+            return ParametersAcceptorSelector::selectSingle(
+                $classType->getMethod($methodName, $scope)->getVariants()
+            )->getParameters();
+        }
+
+        if ($node->name instanceof Expr) {
+            return $this->getParametersForExpr($node->name, $scope);
+        }
+
+        return $this->getParametersForName($node->name, $node->args, $scope);
+    }
+
+    /**
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForExpr(Expr $expr, Scope $scope): array
+    {
+        $type = $scope->getType($expr);
+
+        if (! $type instanceof ParametersAcceptor) {
+            return [];
+        }
+
+        return $type->getParameters();
+    }
+
+    /**
+     * @param Node\Arg[] $args
+     *
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForName(Name $name, array $args, Scope $scope): array
+    {
+        try {
+            return ParametersAcceptorSelector::selectFromArgs(
+                $scope,
+                $args,
+                $this->broker->getFunction($name, $scope)->getVariants()
+            )->getParameters();
+        } catch (FunctionNotFoundException $functionNotFoundException) {
+            return [];
+        }
+    }
+
+    private function processArgument(Expr $expr, Scope $scope): VariableAssignPair
```

Comment:
This looks like `VariableAssignPairFactory`.

*process/refactor* method should change nodes in some way